sdccc/src/main/java/com/draeger/medical/sdccc/messages/MessageStorage.java

```diff
@@ -310,7 +315,45 @@ protected MessageContent convertMessageToMessageContent(final Message message) {
                 mdibVersionGroups,
                 actions,
                 message.getID(),
-                isSOAP);
+                isSOAP,
+                getSender(message));
+    }
+
+    private String getSender(final Message message) {
+        if (message.getDirection() == CommunicationLog.Direction.INBOUND) {
+            final CommunicationContext communicationContext = message.getCommunicationContext();
+            if (communicationContext == null) {
+                // NOTE: this should never happen. If it does, this should be considered a bug.
+                LOG.trace("Encountered message (uuid={}) without a CommunicationContext.", message);
+                testRunObserver.invalidateTestRun("Encountered message without a CommunicationContext.");
+                return null;
+            }
+            final TransportInfo transportInfo = communicationContext.getTransportInfo();
+            if (transportInfo == null) {
+                // NOTE: this should never happen. If it does, this should be considered a bug.
+                LOG.trace("Encountered message (uuid={}) without a TransportInfo.", message);
+                testRunObserver.invalidateTestRun("Encountered message without a TransportInfo.");
+                return null;
+            }
+            final Optional<String> remoteAddress = transportInfo.getRemoteAddress();
+            if (remoteAddress.isEmpty()) {
+                // NOTE: this should never happen. If it does, this should be considered a bug.
+                LOG.trace("Encountered inbound message (uuid={}) without a remoteAddress.", message);
+                testRunObserver.invalidateTestRun("Encountered inbound message without a remoteAddress.");
+                return null;
+            } else {
+                return remoteAddress.orElseThrow();
+            }
+        } else if (message.getDirection() == CommunicationLog.Direction.OUTBOUND) {
+            return DEFAULT_OUTBOUND_SENDER;
```

Comment:
Do not replace the data and only write the original data into the database.

Reply:
Done.